### PR TITLE
Add infrastructure for puppeteer

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Paves the way for migrating our `go_chrome_test` suite to puppeteer (and we can follow with firefox at a later date).

Note: we also verified that running multiple webdriver tests in the same suite (dev_appserver) works fine, bypassing whatever bug happens in Golang; this leads me to the theory that some unconsumed input buffer is getting exhausted.

### CODE REVIEW

File: webdriver/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "wptdashboard-webdriver",
+  "description": "Node packages leveraged by build rules for the wpt.fyi webdriver tests.",
+  "license": "BSD-3-Clause",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/web-platform-tests/wpt.fyi.git"
+  },
+  "devDependencies": {
```
Comment (Hexcles) [inline webdriver/package.json]: Did you forget to save the dependencies? There's no prod or dev dependencies at all in `package.json`.

File: webdriver/dev_appserver.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
+const { spawn } = require('child_process');
```
Comment (Hexcles) [inline webdriver/dev_appserver.js]: Do we need a copyright header?

File: webdriver/webdriver.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+const dev_appserver = require('./dev_appserver');
```
Comment (Hexcles) [inline webdriver/webdriver.js]: ditto

File: webdriver/webdriver.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  test('/results/', async () => {
+    browser = await puppeteer.launch();
+    const page = await browser.newPage();
+    await page.goto(`${server.url}/results/`);
+    await browser.close();
+  }).timeout(10000);
+
+  suiteTeardown(async () => {
+    log('closing dev_appserver...');
+    server.process.kill();
```
Comment (Hexcles) [inline webdriver/webdriver.js]:
```suggestion
    server.close();
```

Just better encapsulation. Furthermore, I think we might as well make `process` a private member.

File: webdriver/dev_appserver.js (hunk 2 above)
Comment (lukebjerring) [reply webdriver/dev_appserver.js]: Done

File: webdriver/webdriver.js (hunk 3 above)
Comment (lukebjerring) [reply webdriver/webdriver.js]: Done

**Pull request merged**